src/main/webapp/script.js

```diff
@@ -128,8 +128,10 @@ async function generateGraph() {
  */
 function getUrl() {
   const depthElem = document.getElementById('num-layers');
+  const nodeName = document.getElementById('node-name') ? document.getElementById('node-name').value || "" : ""; 
```

Comment:
honestly it really shouldn't because it's in the html. I think I just did that because .value on something that's undefined doesn't sit well with JS, but do you think it's unnecessary?

Reply:
Directly calling `.value` would give a better indication of where the error is than if someone accidentally changed the id 'node-name' to something else like 'node-id', right? They would have a hard time tracking down the bug because the failure is silently converted to the default empty string.